### DISCUSSION TITLE
Merging rows

### DISCUSSION THREAD

**Discussion by Shahin-rmz** [saulpw/visidata#1356] [Q&A] [answered]
Hello
I have following table.

col1|col2|col3|col4|col5|
----|----|----|----|----|
Key1| | | | | |
| | value1 | | | |
| | | value2 | | | |
| | | |value3| |
| | | | | value4|
|key2|||||

I want to merge rows in the way it shows normal table
col1|col2|col3|col4|col5|
----|----|----|----|----|
Key1| value1|value2 |value3 |value4 |value5 |
key2| value6|value7|value8|value9|

how is that possible with Visidata?
Thanks

**Comment by saulpw** [comment, marked as answer]
Use `gs` to select all rows, then `f` to fill the nulls in each column with the row above it.Then `gu` to unselect all rows, select all the empty/null values (`g|^$` should work), and `gd` to delete them.  What's left should be the full rows that you want.

Note that you may have to set `options.null_value=""` so that VisiData considers empty strings null values; or you can `Shift+M` to melt the sheet, go to an empty cell and press `,` to select all empty cells, then `gzd` to set them to null values).

**Comment by Shahin-rmz** [comment]
Thanks for your response.
As you said, I melted the sheet `SHIFT + m` and set the empty cells to null with `,` .
Here is how it looks like:
![image](https://user-images.githubusercontent.com/32805117/164100266-f750da3e-d417-4bee-b324-9ee0bbf7979e.png)

then `gs` and `f`



and this is how the results looks like.
![image](https://user-images.githubusercontent.com/32805117/164103144-16c13f1e-dd31-480b-be3b-6d471df67587.png)

However, when I select null values with `g|^$` and then `gd`, I lost first 1000 rows.

Would you please take a look at [raw data](https://github.com/saulpw/visidata/files/8516537/Good2.csv)
and also the data [after](https://github.com/saulpw/visidata/files/8516542/Saul.csv) doing your guide and tell me what I have done wrong?

These are multiple choice questions and their answers of an exam and some questions (first column) are multiple lines (multiple rows). This makes me a bit confused.
Maybe it would be better to merge the questions into one row and then do your manual.
bit confusing.
Thanks.

**Reply by saulpw** [reply]
The `g|^$` command selects all rows with at least one cell with empty contents.  So I guess your first 1000 rows have that extra column with null values.  You can either remove those columns, or find a different way to select the partially-filled rows.

**Comment by Shahin-rmz** [comment]
@saulpw I was able to make the project done, with Pandas, (in Jupyter).
![image](https://user-images.githubusercontent.com/32805117/164562392-e45b573a-e135-4b83-8dcf-040249aa8b50.png)

However I was thinking with myself, it would be great if we could have **interactive DataFrame** of pandas which we can work in VisiData.
I think it was mentioned before about being able to write python code inside VisiData, but this one is just pure **live DataFrame**.
I think it brings VisiData to another level.
Best

**Reply by saulpw** [reply]
Hi @Shahin-rmz, try using `vd.view_pandas(df)`, which shows the pandas dataframe within VisiData.

**Reply by Shahin-rmz** [reply]
How about having [live](https://github.com/saulpw/visidata/discussions/1294#discussioncomment-2278164) pandas dataframe and be able to do advanced python on it?
Use the true power of VisiData and Python together
Check [this](https://pypi.org/project/pandasgui/) out.
